lib/Rest/VstResponse.cpp

```diff
@@ -98,18 +134,42 @@ void VstResponse::addPayload(VPackBuffer<uint8_t>&& buffer,
       VPackBuilder builder(tmpBuffer, options);
       VelocyPackHelper::sanitizeNonClientTypes(input, VPackSlice::noneSlice(),
                                                builder, options, true, true, true);
-      if (_payload.empty()) {
-        _payload = std::move(tmpBuffer);
+      if (_contentType == rest::ContentType::VPACK) {
+        if (_payload.empty()) {
+          _payload = std::move(tmpBuffer);
+        } else {
+          _payload.append(tmpBuffer.data(), tmpBuffer.size());
+        }
+        return;
+      } else if (_contentType == rest::ContentType::JSON) {
+        VPackSlice finalSlice(tmpBuffer.data());
+        StringBuffer plainBuffer;
+        arangodb::basics::VelocyPackDumper dumper(&plainBuffer, options);
+        dumper.dumpValue(finalSlice);
+        _payload.reset();
+        _payload.append(plainBuffer.data(), plainBuffer.length());
```

Comment:
mmmh, there is no `return` here anymore.
doesn't this mean that below the same stuff will be added to `_payload` again?